jax/_src/lax/linalg.py

```diff
@@ -2417,6 +2417,26 @@ def _tridiagonal_solve_gpu_lowering(lowering, ctx, dl, d, du, b, *, m, n, ldb, t
       b_shape_vals=b_shape_vals)]
 
 
+def _tridiagonal_product(dl, d, du, b):
+  y = lax.reshape(d, d.shape + (1,)) * b
+  y = y.at[..., 1:, :].add(dl[..., 1:, None] * b[..., :-1, :])
+  y = y.at[..., :-1, :].add(du[..., :-1, None] * b[..., 1:, :])
+  return y
```

Comment:
We don't have an existing primitive for computing tridiagonal products, so here's my version. Suggestions for improvements are welcome!

Reply:
I wonder if it'd be faster on GPUs and TPUs to do some dense block matrix multiplies here. Or it might be a place we could use a Pallas kernel. We'd have to research or test that, and I'm not sure it's worthwhile, so let's not do it now, but I thought I'd mention it for posterity.